ext/drx/drx.c

```diff
@@ -219,32 +219,13 @@ DR_EXPORT
 bool
 drx_aflags_are_dead(instr_t *where)
 {
-    instr_t *instr;
-    uint flags;
-    for (instr = where; instr != NULL; instr = instr_get_next(instr)) {
-        /* we treat syscall/interrupt as aflags read */
-        if (instr_is_syscall(instr) || instr_is_interrupt(instr))
-            return false;
-        flags = instr_get_arith_flags(instr, DR_QUERY_DEFAULT);
-        if (TESTANY(EFLAGS_READ_ARITH, flags))
-            return false;
-        if (TESTALL(EFLAGS_WRITE_ARITH, flags))
-            return true;
-        if (instr_is_cti(instr)) {
-            if (instr_is_app(instr) &&
-                (instr_is_ubr(instr) || instr_is_call_direct(instr))) {
-                instr_t *next = instr_get_next(instr);
-                opnd_t tgt = instr_get_target(instr);
-                /* continue on elision */
```

Comment:
Comparing this old implementation with `drreg_forward_analysis`: I see that the latter doesn't have this logic to continue on elision. It just bails on seeing a cti: https://github.com/DynamoRIO/dynamorio/blob/51e5f9c946c566e704e133c4b784f1547dfd3328/ext/drreg/drreg.c#L770

To me it seems a minor optimisation, and it should be fine to not have it anymore. But better to consult @derekbruening to confirm if this optimisation has any other significance.

Reply:
This is for -max_elide_{jmp,call} > 0 where DR makes a superblock by merging the target into the source block.  It's turned off for -code_api by default b/c most clients don't want it.

Reply:
I guess then it should be safe for aflags liveness to not recognise such elision cases, as we don't expect `drx_aflags_are_dead` to be invoked if code manipulation apis are disabled.